aws/resource_aws_ec2_capacity_reservation.go

```diff
@@ -0,0 +1,278 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceAwsEc2CapacityReservation() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsEc2CapacityReservationCreate,
+		Read:   resourceAwsEc2CapacityReservationRead,
+		Update: resourceAwsEc2CapacityReservationUpdate,
+		Delete: resourceAwsEc2CapacityReservationDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"availability_zone": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"ebs_optimized": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+				Default:  false,
+			},
+			"end_date": {
```

Comment:
We have an upstream validation function which can help surface issues at plan time for this attribute. 😄 

```
ValidateFunc: validation.ValidateRFC3339TimeString,
```